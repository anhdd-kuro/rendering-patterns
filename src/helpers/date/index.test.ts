import { to2DigitsMonthString, DateFormat, to2DigitsDateString, toString } from ".";

describe("Date helper", () => {
  describe("to2DigitsMonthString", () => {
    it("should convert correctly", async () => {
      const input = new Date("2022-05-14");
      const result = to2DigitsMonthString(input);
      expect(result).equal("05");
      expect(result).toMatchSnapshot();
    });
  });

  describe("to2DigitsDateString", () => {
    it("should convert correctly", async () => {
      const input = new Date("2022-05-14");
      const result = to2DigitsDateString(input);
      expect(result).equal("14");
      expect(result).toMatchSnapshot();
    });
  });

  describe("toString", () => {
    (
      [
        {
          date: new Date("2022-05-14"),
          format: "yyyy-mm-dd",
          result: "2022-05-14",
        },
        {
          date: new Date("2022-05-14"),
          format: "yyyy/mm/dd",
          result: "2022/05/14",
        },
      ] as { date: Date; format: DateFormat; result: string }[]
    ).forEach(({ date, format, result: expectedResult }) => {
      it(`format: ${format} should convert to ${expectedResult}`, async () => {
        const result = toString(date, format);
        expect(result).equal(expectedResult);
        expect(result).toMatchSnapshot();
      });
    });
  });
});
