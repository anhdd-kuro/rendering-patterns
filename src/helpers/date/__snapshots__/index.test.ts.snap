// Vitest Snapshot v1

exports[`Date helper > to2DigitsDateString > should convert correctly 1`] = `"14"`;

exports[`Date helper > to2DigitsMonthString > should convert correctly 1`] = `"05"`;

exports[`Date helper > toString > format: yyyy/mm/dd should convert to 2022/05/14 1`] = `"2022/05/14"`;

exports[`Date helper > toString > format: yyyy-mm-dd should convert to 2022-05-14 1`] = `"2022-05-14"`;
